### ISSUE TITLE
[$500]  Request Money - Conversation is not scrolled to the bottom when request money

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#22367]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Navigate to a 1:1 conversation
2. Scroll a moderate amount of the conversation history so the latest message is not visible
3. On the compose box - Click the + button
4. Click on request money and finish the IOU flow

## Expected Result:
The conversation should be scrolled to the bottom and the latest message (IOU) should be visible

## Actual Result:
The conversation is not scrolled to the bottom when Request Money

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.37.2

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/85ed47ed-b5ba-4b6a-8edf-961a65829111

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~016ec292c1dce29715</li>
        <li>Upwork Job ID: 1677287906884644864</li>
        <li>Last Price Increase: 2023-09-12</li>
    </ul>
</details>

**Comment by PrashantMangukiya** [comment]
## Proposal

Posting proposal early as per new [guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#propose-a-solution-for-the-job)

### Please re-state the problem that we are trying to solve in this issue.
Conversation is not scrolled to the bottom when request money

### What is the root cause of that problem?
We are not scrolling chat to bottom during request money as shown below.

https://github.com/Expensify/App/blob/35e8d4643eb5f1ecd4db78bac6e145585e3d333a/src/pages/iou/steps/MoneyRequestConfirmPage.js#L198

https://github.com/Expensify/App/blob/35e8d4643eb5f1ecd4db78bac6e145585e3d333a/src/pages/iou/steps/MoneyRequestConfirmPage.js#L111-L154

### What changes do you think we should make in order to solve the problem?
We have to scroll to bottom during confirm as shown below:
```javascript
onConfirm={selectedParticipants => {
    createTransaction(selectedParticipants);
    ReportActionComposeFocusManager.focus()
}}
```

### What alternative solutions did you explore? (Optional)
We can also scroll to bottom within createTransaction as shown below:
```javascript
const createTransaction = useCallback(
    (selectedParticipants) => {
        const trimmedComment = props.iou.comment.trim();

        // IOUs created from a group report will have a reportID param in the route.
        // Since the user is already viewing the report, we don't need to navigate them to the report
        if (iouType.current === CONST.IOU.MONEY_REQUEST_TYPE.SPLIT && CONST.REGEX.NUMBER.test(reportID.current)) {
            IOU.splitBill(
                ...
            );
            ReportActionComposeFocusManager.focus() // *** ADD THIS ***
            return;
        }

        // If the request is created from the global create menu, we also navigate the user to the group report
        if (iouType.current === CONST.IOU.MONEY_REQUEST_TYPE.SPLIT) {
            IOU.splitBillAndOpenReport(
                ...
            );
            ReportActionComposeFocusManager.focus() // *** ADD THIS ***
            return;
        }

        IOU.requestMoney(
            ...
        );
        ReportActionComposeFocusManager.focus() // *** ADD THIS ***
    },
    [props.iou.amount, props.iou.comment, props.currentUserPersonalDetails.login, props.currentUserPersonalDetails.accountID, props.iou.currency, props.report],
);
```

**Comment by s77rt** [comment]
This issue is for Android/Native only right? Asking because the issue description says Android/Chrome? Assuming it's on Android/Native only, I think we can put this one on hold for https://github.com/Expensify/App/issues/15964. The root cause seems that calling the scroll function while another animation is ongoing will do nothing. We have this commit https://github.com/facebook/react-native/commit/681b35daab2d0443278fe18c364b0e72c8c85673 that aborts the ongoing animation before making a new one. It may fix this issue, it would be glad if anyone can test that.

**Comment by NicMendonca** [comment]
@allroundexperts any thoughts here?

**Comment by NicMendonca** [comment]
@allroundexperts any feedback here?

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by NicMendonca** [comment]
>  Issue not reproducible during KI retests. (First week)



closing

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by lanitochka17** [comment]
Issue reproducible on Build 1.3.60-1
mWeb/Chrome


https://github.com/Expensify/App/assets/78819774/37d5fa22-ec0d-472c-95d9-e94181f74fc6

**Comment by NicMendonca** [comment]
@allroundexperts do you have any feedback on this proposal[ here](https://github.com/Expensify/App/issues/22367#issuecomment-1639716302)?

**Comment by NicMendonca** [comment]
@madmax330 should I close this?

**Comment by madmax330** [comment]
I think we close after 4 weeks of not being reproducible

**Comment by NicMendonca** [comment]
@tienifr https://github.com/Expensify/App/issues/27724 is fixed though!

@mvtglobally can you reproduce this?

**Comment by NicMendonca** [comment]
bump bump @mvtglobally

**Comment by mvtglobally** [comment]
Will provide feedback in a little bit

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Fourth week) @NicMendonca

**Comment by lanitochka17** [comment]
Issue reproducible on latest build 1.3.88-3

https://github.com/Expensify/App/assets/78819774/1d9f0280-ff63-4f16-81f8-38878e5344d2

**Comment by NicMendonca** [comment]
@madmax330 or maybe a regression re-caused this? 🤔 (since it was fixed for quite a while) Let me know how you think we should proceed!

**Comment by madmax330** [comment]
Yes we're modifying a lot for the money 2020 and other waves right now, so I wouldn't be surprised
